src/app/digital/views/components/ButtonView.ts

```diff
@@ -0,0 +1,56 @@
+import {SVGDrawing} from "svg2canvas";
+
+import {SELECTED_FILL_COLOR} from "core/utils/Constants";
+
+import {V} from "Vector";
+
+import {Images} from "core/utils/Images";
+
+import {Button} from "core/models/types/digital";
+
+import {Signal} from "digital/models/sim/Signal";
+
+import {RenderInfo}             from "core/views/BaseView";
+import {PressableComponentView} from "core/views/PressableComponentView";
+
+import {DigitalViewInfo} from "../DigitalViewInfo";
+
+
+export class ButtonView extends PressableComponentView<Button, DigitalViewInfo> {
+    public static BOUNDING_HEIGHT = 1.54;
+
+    protected onImg: SVGDrawing;
+    protected offImg: SVGDrawing;
+
+    public constructor(info: DigitalViewInfo, obj: Button) {
+        super(info, obj, V(1, 1), V(1, 1));
+
+        this.onImg  = Images.GetImage("buttonDown.svg");
+        this.offImg = Images.GetImage("buttonUp.svg");
+    }
+
+    public override onPropChange(propKey: string): void {
+        super.onPropChange(propKey);
+
+    }
+
+    // TODO: move this to a SwitchController or something?
+    //  it also should be causing a propagation change, not an image change
+    //  and the image will be calculated based on the propagation
+    public override onPress(): void {
```

Comment:
`onPress` should just set the state to `Signal.On` and `onRelease` should set it to `Signal.Off`